lib/src/menu_item.dart

```diff
@@ -33,10 +33,12 @@ class MenuItem {
     this.toolTip,
     this.icon,
     this.disabled = false,
-    this.submenu,
+    List<MenuItem> items = const [],
+    Menu? submenu,
     this.onClick,
   })  : id = _generateMenuItemId(),
-        type = 'submenu';
+        type = 'submenu',
+        submenu = (submenu?..items.addAll(items)) ?? Menu(items: items);
```

Comment:
I don't think items should be added to submenu 's items , if both parameters are set, the combined menu will be confusing

Reply:
I'm considering whether to remove the submenu and keep only the items.

```dart
Menu(
  items: [
    MenuItem(),
    MenuItem.submenu(
      items: [
        MenuItem(),
        MenuItem(),
      ],
    )
  ],
);
```

Reply:
Would you internally still create a `Menu` to wrap the `items`?
If no I think that this would be a good call; if yes, I don't think it would be worth doing a breaking change for.

IMO the most important part would be to make the `items` in `Menu` non-nullable, to prevent the crashes. I could remove the `submenus` or `items` thing from this patch and make a seperate one for that